### PR TITLE
fix(text): fix horizontal and vertical alignment error

### PR DESCRIPTION
**Pull request by singlecoder**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by icyfe — APPROVED** [review]

**Review by gz65555 — APPROVED** [review]

File: packages/core/src/2d/text/TextRenderer.ts [hunk 1]
```diff
@@ -378,7 +378,23 @@ export class TextRenderer extends Renderer {
...
+      const diffPivotX = ((originWidthInPixel - width) * 0.5) / width;
+      pivot.x = 0.5 + diffPivotX * (horizontalAlignment === TextHorizontalAlignment.Left ? 1 : -1);
+    }
+    if (originHeightInPixel > height && verticalAlignment !== TextVerticalAlignment.Center) {
+      const diffPivotY = ((originHeightInPixel - height) * 0.5) / height;
+      pivot.y = 0.5 + diffPivotY * (verticalAlignment === TextVerticalAlignment.Top ? -1 : 1);
+    }
+    sprite.pivot = pivot;
+
     // If add fail, set texture for sprite.
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/2d/text/TextRenderer.ts:398]:
```
    const { pixelsPerUnit, pivot } = sprite;
    switch (horizontalAlignment) {
      case TextHorizontalAlignment.Left:
        pivot.x = (this.width * pixelsPerUnit) / width / 2;
        break;
      case TextHorizontalAlignment.Right:
        pivot.x = 1 - (this.width * pixelsPerUnit) / width / 2;
        break;
      case TextHorizontalAlignment.Center:
        pivot.x = 0.5;
        break;
      default:
        break;
    }
    switch (verticalAlignment) {
      case TextVerticalAlignment.Top:
        pivot.y = 1 - (this.height * pixelsPerUnit) / height / 2;
        break;
      case TextVerticalAlignment.Bottom:
        pivot.y = (this.height * pixelsPerUnit) / height / 2;
        break;
      case TextVerticalAlignment.Center:
        pivot.y = 0.5;
        break;
      default:
        break;
    }
    sprite.pivot = pivot;
```

Comment (cptbtptpbcptdtptp) [reply packages/core/src/2d/text/TextRenderer.ts:398]: image

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Review by GuoLei1990 — APPROVED** [review]
+1

**Pull request merged**